src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -407,6 +432,8 @@ pub struct BinaryEExpWriter_1_1<'value, 'top> {
     allocator: &'top BumpAllocator,
     buffer: &'value mut BumpVec<'top, u8>,
     value_writer_config: ValueWriterConfig,
+    macros: &'value MacroTable,
+    signature_iter: SignatureIterator<'value>,
```

Comment:
🪧 Common logic for iterating over the parameters in a macro signature are encapsulated in the `SignatureIterator` type, which is now nested in both the binary and text e-expression writers.